tracer/src/Datadog.Trace/Debugger/Expressions/ProbeExpressionParser.cs

```diff
@@ -0,0 +1,483 @@
+                            }
+
+                            if (readerValue == "@return")
+                            {
+                                return itParameter;
+                            }
+
+                            if (readerValue == "@duration")
+                            {
+                                return itParameter;
+                            }
+
+                            if (readerValue == "@it")
+                            {
+                                // current item in iterator
+                                if (itParameter == null)
+                                {
+                                    AddError(readerValue, "current item in iterator is null");
+                                    return Expression.Parameter(UndefinedValueType, Expressions.UndefinedValue.Instance.ToString());
+                                }
+
+                                return itParameter;
+                            }
+
+                            if (readerValue == "@exceptions")
+                            {
+                                return itParameter;
+                            }
+
+                            return Expression.Constant(readerValue);
+                        }
+
+                    case JsonToken.Integer:
+                        {
+                            return Expression.Constant(Convert.ChangeType(readerValue, TypeCode.Int32));
+                        }
+
+                    case JsonToken.StartArray:
+                        {
+                            _arrayStack++;
+                            break;
+                        }
+
+                    case JsonToken.EndArray:
+                        {
+                            _arrayStack--;
+                            break;
+                        }
+
+                    case JsonToken.Null:
+                        {
+                            return Expression.Constant(null);
+                        }
+                }
+            }
+            catch (Exception e)
+            {
+                AddError(reader.Value?.ToString() ?? "N/A", e.Message);
+                return ReturnDefaultValueExpression();
+            }
+        }
+
+        return null;
+    }
+
+    private void AddError(string expression, string error)
+    {
+        (_errors ??= new List<EvaluationError>()).Add(new EvaluationError { Expression = expression, Message = error });
+    }
+
+    private bool ConditionalRead(JsonTextReader reader, bool shouldAdvanceReader)
+    {
+        return !shouldAdvanceReader || reader.Read();
+    }
+
+    private void SetReaderAtExpressionStart(JsonTextReader reader)
+    {
+        while (reader.Read())
+        {
+            if (reader.TokenType != JsonToken.PropertyName)
+            {
+                continue;
+            }
+
+            if ((reader.Value?.ToString() == "json"))
+            {
+                reader.Read();
+                reader.Read();
```

Comment:
Magic magic. I'd suggest to add a comment explaining why we jump exactly two tokens.